packages/react/src/primitives/SwitchField/useSwitch.ts

```diff
@@ -18,7 +18,9 @@ interface UseSwitch {
 export const useSwitch = (props: UseSwitchProps): UseSwitch => {
   const { onChange, isChecked, defaultChecked, isDisabled } = props;
   const isControlled = typeof isChecked !== 'undefined';
-  const [isOn, setIsOn] = useState(isControlled ? isChecked : defaultChecked);
+  const [isOn, setIsOn] = useState(
+    isControlled ? isChecked : defaultChecked ?? false
```

Comment:
Rather than use nullish coalescing here, would it be possible to coerce `defaultChecked` to a boolean?